web/modules/custom/joinup_core/src/NodeWorkflowAccessControlHandler.php

```diff
@@ -138,6 +139,18 @@ public function entityAccess(EntityInterface $entity, $operation, AccountInterfa
       case 'delete':
         return $this->entityDeleteAccess($entity, $account);
 
+      case 'post comments':
+        $parent_state = $this->relationManager->getParentState($entity);
+        if ($parent_state === 'archived') {
```

Comment:
This is pretty self-explanatory, but still I think it's a good idea to provide a comment for every switch. E.g.

    // Don't allow to post comments to archived collections.